docs/changes/io.fits/7312.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+A compressed image HDU created from the header of a PRIMARY HDU, now correctly updates
```

Comment:
Please rename this fragment to match the PR number, not the original issue number. And then squash the commits. Thanks!

Reply:
Oh gosh ... right ... sorry for that... :(